samples/kvssink_intermittent_sample.cpp

```diff
@@ -0,0 +1,323 @@
+        if(cv.wait_for(lck, std::chrono::seconds(KVS_INTERMITTENT_PLAYING_INTERVAL_SECONDS)) != std::cv_status::timeout) {
+            break;
+        }
+
+        LOG_INFO("[KVS sample] Stopping stream to KVS for " << KVS_INTERMITTENT_PAUSED_INTERVAL_SECONDS << " seconds");
+
+        // EOS event pushes frames buffered by the h264enc element down to kvssink.
+        GstEvent* eos = gst_event_new_eos();
+        gst_element_send_event(pipeline, eos);
+
+        // Wait for the EOS event to return from kvssink to the bus which means all elements are done handling the EOS.
+        // We don't want to flush until the EOS is done to ensure all frames buffered in the pipeline have been processed.
+        cv.wait(lck);
+
+        // Set videotestsrc to paused state because it does not stop producing frames upon EOS,
+        // and the frames are not cleared upon flushing.
+        if(STRCMPI(GST_ELEMENT_NAME(source), KVS_GST_TEST_SOURCE_NAME) == 0) {
+            gst_element_set_state(source, GST_STATE_PAUSED);
+        }
+
+        // Flushing to remove EOS status.
+        GstEvent* flush_start = gst_event_new_flush_start();
+        gst_element_send_event(pipeline, flush_start);
+
+        // Using cv.wait_for to break sleep early upon signal interrupt. Checking for termination again before waiting.
+        if(terminated || cv.wait_for(lck, std::chrono::seconds(KVS_INTERMITTENT_PAUSED_INTERVAL_SECONDS)) != std::cv_status::timeout) {
+            break;
+        }
+
+        LOG_INFO("[KVS sample] Starting stream to KVS for " << KVS_INTERMITTENT_PLAYING_INTERVAL_SECONDS << " seconds");
+        
+        // Stop the flush now that we are resuming streaming.
+        GstEvent* flush_stop = gst_event_new_flush_stop(true);
+        gst_element_send_event(pipeline, flush_stop);
+
+        // Set videotestsrc back to playing state.
+        if(STRCMPI(GST_ELEMENT_NAME(source), KVS_GST_TEST_SOURCE_NAME) == 0) {
+            gst_element_set_state(source, GST_STATE_PLAYING);
+        }
+    }
+    LOG_DEBUG("[KVS sample] Exited stopStartLoop");
+}
+
+int main(int argc, char *argv[])
+{
+    signal(SIGINT, sigint_handler);
+
+    CustomData customData;
+    GstElement *pipeline, *source, *clock_overlay, *video_convert, *source_filter, *encoder, *sink_filter, *kvssink;
+    GstCaps *source_caps, *sink_caps;
+    GstBus *bus;
+    guint bus_watch_id;
+    StreamSource source_type;
+    char stream_name[MAX_STREAM_NAME_LEN + 1] = {0};
+
+    gst_init(&argc, &argv);
+
+
+    /* Parse input arguments */
+
+    // Check for invalid argument count, get stream name.
+    if(argc > 3) {
+        LOG_ERROR("[KVS sample] Invalid argument count, too many arguments.");
+        LOG_INFO("[KVS sample] Usage: " << argv[0] << " <streamName (optional)> <testsrc or devicesrc (optional)>");
+        return -1;
+    } else if(argc > 1) {
+        STRNCPY(stream_name, argv[1], MAX_STREAM_NAME_LEN);
+        stream_name[MAX_STREAM_NAME_LEN] = '\0';
```

Comment:
Is this needed? You already set this on line 181.

Reply:
Correct, not needed now that the whole buffer is being initialized as null. Will remove